### DISCUSSION TITLE
Feature Request: Optional 'auto disable' of buttons during async on_click callbacks

### DISCUSSION THREAD

**Discussion by duane-space** [zauberzeug/nicegui#1864] [Ideas / Feature Requests]
I commonly include a button.disable() and button.enable() at the start/end of my on_click callbacks. There is an [example](https://nicegui.io/documentation/button#disable_button_with_a_context_manager) in the documentation of doing this more intelligently with a context manager. I think adding this directly to the API would be even better.

This has been discussed before, when the aforementioned example was added to the documentation. https://github.com/zauberzeug/nicegui/pull/1238

At the time, valid concern about making it the _default_ functionality was brought up. So I think the proposed implementation would be adding an _optional_ boolean 'auto_disable' to the ui.button() constructor that defaults to off.

`ui.button(auto_disable=False,...)`

Edit:

Oh, was also some relevant discussion here:
https://github.com/zauberzeug/nicegui/discussions/1220

**Comment by rodja** [comment]
Oh, we missed our chance to make it the default behaviour in 1.4 (see https://github.com/zauberzeug/nicegui/pull/1238#issuecomment-1649124226). But I still like the idea. Let's do it in 1.5.

**Reply by falkoschindler** [reply]
I'm still not sure about the right default behavior.
But for now we can disable it by default and introduce the new parameter in 1.4.x. No need to wait for 1.5.
Would you like to create a pull request, @duane-impulse? 🙂
